### PR TITLE
Add JMS client Ack test for inbound endpoint

### PR DESCRIPTION
**Pull request by a5anka**

### CODE REVIEW

File: integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    /**
+     * Test whether consuming message from a queue using client ack mode works
+     *
+     * @throws Exception if any error occurred while running tests
+     */
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+    @Test(groups = {"wso2.esb"}, description = "Test JMS client ack mode with inbound endpoint")
+    public void testJmsQueueToHttpWithInboundEndpoint() throws Exception {
+        LogViewerClient logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
```
Comment (himasha) [inline integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java]: Shall we move this to initialize() ?

File: integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        sendMessage();
+
+        //check for the log
+        boolean assertValue = Utils.assertIfSystemLogContains(logViewerClient,
+                                                              "** jmsClientAckInboundEPSendInSequence was called **");
+
+        Assert.assertTrue(assertValue, "Message was not received to the inbound EP in client ack mode.");
+        Assert.assertTrue(Utils.isQueueEmpty(QUEUE_NAME),"Queue should be empty if ack was properly sent");
+    }
+
```
Comment (Subasinghe) [inline integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java]: Remove unnecessary new lines.

File: integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java (hunk 1 above)
Comment (a5anka) [reply integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java]: No we can't it needs to be enabled and disabled within the scope of the test method. Other wise there can be unnecessary polling.

File: integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java (hunk 2 above)
Comment (a5anka) [reply integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java]: :+1: will do

Comment (a5anka) [reply integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java]: Fixed in current version

File: integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/Utils.java [hunk 3]
```diff
@@ -21,6 +21,8 @@
 import org.apache.axiom.om.OMElement;
```
Comment (srsharon) [inline integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/Utils.java]: Shall we update the license. it should the the year created only.

File: integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                "      <ser:request>" +
+                "        <xsd:symbol>IBM</xsd:symbol>" +
+                "      </ser:request>" +
+                "    </ser:getQuote>" +
+                "  </soapenv:Body>" +
+                "</soapenv:Envelope>";
+        try {
+            sender.connect(QUEUE_NAME);
+            sender.pushMessage(message);
+            Thread.sleep(2000);
```
Comment (nuwanw) [inline integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java]: can we remove the sleep and run the assertion in a loop with max timeout

File: integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/Utils.java (hunk 3 above)
Comment (a5anka) [reply integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/Utils.java]: Fixed

**Pull request merged**